configstack/module.go

```diff
@@ -52,7 +53,68 @@ func ResolveTerraformModules(terragruntConfigPaths []string, terragruntOptions *
 		return []*TerraformModule{}, err
 	}
 
-	return crosslinkDependencies(mergeMaps(modules, externalDependencies), canonicalTerragruntConfigPaths)
+	crossLinkedModules, err := crosslinkDependencies(mergeMaps(modules, externalDependencies), canonicalTerragruntConfigPaths)
+	if err != nil {
+		return []*TerraformModule{}, err
+	}
+
+	finalModules := removeExcludedDirs(crossLinkedModules, terragruntOptions)
+
+	return finalModules, nil
+}
+
+// removeExcludedDirs iterates over a module slice and removes all entries, which should be ignored via the terragrunt-exclude-dir CLI flag.
+// Modules are also removed if one of its dependencies are to be excluded. Returns the cleaned-up slice of modules.
+func removeExcludedDirs(modules []*TerraformModule, terragruntOptions *options.TerragruntOptions) []*TerraformModule {
```

Comment:
Consider adding unit tests just for this method. You could pass in your own mock set of modules with various paths and see what it returns.

Reply:
I already added to unit tests for `ResolveTerraformModules` which calls `removeExcludedDirs`.
The mock set was especially trimmed for excluded dirs. However I can add even more tests if you would like.